17.1.0 (#337)
MetaMask/providers: CHANGELOG.md

## [Unreleased]

## [17.1.0]

> ### Uncategorized
> 
> - Replace `tsup` with `ts-bridge` as build tool ([#336](https://github.com/MetaMask/providers/pull/336))
> - Bump @metamask/json-rpc-middleware-stream from 7.0.1 to 8.0.0 ([#327](https://github.com/MetaMask/providers/pull/327))
> - Bump @metamask/json-rpc-engine from 8.0.2 to 9.0.0 ([#328](https://github.com/MetaMask/providers/pull/328))
> - refactor: Use named `extension-port-stream` import for Vite compatibility ([#331](https://github.com/MetaMask/providers/pull/331))
> - Bump @metamask/utils from 8.4.0 to 8.5.0 ([#329](https://github.com/MetaMask/providers/pull/329))
> - Bump @metamask/rpc-errors from 6.2.1 to 6.3.0 ([#326](https://github.com/MetaMask/providers/pull/326))
> - Bump ws from 8.13.0 to 8.17.1 in the npm_and_yarn group across 1 directory ([#333](https://github.com/MetaMask/providers/pull/333))
> - Bump braces from 3.0.2 to 3.0.3 in the npm_and_yarn group across 1 directory ([#332](https://github.com/MetaMask/providers/pull/332))
> - chore(devdeps): @lavamoat/allow-scripts@^3.0.0->^3.0.4 ([#322](https://github.com/MetaMask/providers/pull/322))

Reviewer: ```suggestion
### Changed

- Bump `@metamask/json-rpc-middleware-stream` from `^7.0.1` to `^8.0.0` ([#327](https://github.com/MetaMask/providers/pull/327))
- Bump `@metamask/json-rpc-engine` from `^8.0.2` to `^9.0.0` ([#328](https://github.com/MetaMask/providers/pull/328))
- Bump `extension-port-stream` from `^3.0.0` to `^4.1.0` ([#331](https://github.com/MetaMask/providers/pull/331))
- Bump `@metamask/utils` from `^8.4.0` to `^8.5.0` ([#329](https://github.com/MetaMask/providers/pull/329))
- Bump `@metamask/rpc-errors` from `^6.2.1` to `^6.3.0` ([#326](https://github.com/MetaMask/providers/pull/326))

### Fixed
- Produce and export ESM-compatible TypeScript type declaration files in addition to CommonJS-compatible declaration files ([#336](https://github.com/MetaMask/providers/pull/336))
  - This fixes the issue of this package being unusable by any TypeScript project that uses `Node16` or `NodeNext` as its `moduleResolution` option.
  - Previously, this package shipped with only one variant of type declaration files, and these files were only CommonJS-compatible, and the `exports` field in `package.json` linked to these files. This is an anti-pattern and was rightfully flagged by the ["Are the Types Wrong?"](https://arethetypeswrong.github.io/) tool as ["masquerading as CJS"](https://github.com/arethetypeswrong/arethetypeswrong.github.io/blob/main/docs/problems/FalseCJS.md). All of the ATTW checks now pass.
- Remove chunk files ([#336](https://github.com/MetaMask/providers/pull/336)).
  - Previously, the build tool we used to generate JavaScript files extracted common code to "chunk" files. While this was intended to make this package more tree-shakeable, it also made debugging more difficult for our development teams. These chunk files are no longer present.
- Use named `extension-port-stream` import to ensure compatibility with bundlers such as Vite ([#331](https://github.com/MetaMask/providers/pull/331))
```

## [Unreleased]

## [17.1.0]

### Changed

- Bump `@metamask/json-rpc-middleware-stream` from `^7.0.1` to `^8.0.0` ([#327](https://github.com/MetaMask/providers/pull/327))
- Bump `@metamask/json-rpc-engine` from `^8.0.2` to `^9.0.0` ([#328](https://github.com/MetaMask/providers/pull/328))
- Bump `extension-port-stream` from `^3.0.0` to `^4.1.0` ([#331](https://github.com/MetaMask/providers/pull/331))
- Bump `@metamask/utils` from `^8.4.0` to `^8.5.0` ([#329](https://github.com/MetaMask/providers/pull/329))
- Bump `@metamask/rpc-errors` from `^6.2.1` to `^6.3.0` ([#326](https://github.com/MetaMask/providers/pull/326))

> ### Fixed

Reviewer: ```suggestion
### Fixed

```

## [Unreleased]

> ## [17.1.0]

Reviewer 2: Are we considering the ESM fixes as a feature? Otherwise should this be 17.0.2 ... ?

Reviewer: I thought adding ESM support (and CJS support in the case of `NodeNext` projects) might count as a feature rather than a regular bugfix. If not, this should probably be rolled into https://github.com/MetaMask/providers/pull/334.

Reviewer 2: Ah, I guess in this case we have a fix that's also kind of a feature. Hmm. Okay, I'm good with 17.1.0 in that case.

### Changed

- Bump `@metamask/json-rpc-middleware-stream` from `^7.0.1` to `^8.0.0` ([#327](https://github.com/MetaMask/providers/pull/327))
- Bump `@metamask/json-rpc-engine` from `^8.0.2` to `^9.0.0` ([#328](https://github.com/MetaMask/providers/pull/328))
- Bump `extension-port-stream` from `^3.0.0` to `^4.1.0` ([#331](https://github.com/MetaMask/providers/pull/331))
- Bump `@metamask/utils` from `^8.4.0` to `^8.5.0` ([#329](https://github.com/MetaMask/providers/pull/329))

> - Bump `@metamask/rpc-errors` from `^6.2.1` to `^6.3.0` ([#326](https://github.com/MetaMask/providers/pull/326))

Reviewer 3: I had thought it was our practice to ignore dependency updates in the changelog. I do have a hard time imagning how it could be relevant to consumers unless the update would be breaking by some other measure. @MajorLift @mcmire what's your thinking about this?

Reviewer 2: It's been our practice to ignore dev dependencies, but not non-dev dependencies. Perhaps this is something we only do in the `core` repo and not elsewhere — but my thought is that it's useful to be able to look back at when a dependency was intentionally introduced or upgraded in case it causes an issue that we weren't aware of later on. Do you think this is information that can be gleaned elsewhere if need be?

Reviewer: Adding to the above, on core, we have a `reviewing-release-prs` doc that specifies dependency bumps as needing to be captured in the changelog.

- https://github.com/MetaMask/core/blob/main/docs/reviewing-release-prs.md#0-tldr--if-youve-done-this-sort-of-thing-before
- https://github.com/MetaMask/core/blob/main/docs/reviewing-release-prs.md#3-review-types-of-changes

I think it's useful information to record in principle, but can get a bit noisy. Being able to record only the "important" bumps would be nice, but I guess to achieve the benefit @mcmire pointed out -- looking back to find the source of an error we didn't know about beforehand -- we can't pick and choose which bumps to record.

### Fixed

- Produce and export ESM-compatible TypeScript type declaration files in addition to CommonJS-compatible declaration files ([#336](https://github.com/MetaMask/providers/pull/336))
  - This fixes the issue of this package being unusable by any TypeScript project that uses `Node16` or `NodeNext` as its `moduleResolution` option.

>   - Previously, this package shipped with only one variant of type declaration files, and these files were only CommonJS-compatible, and the `exports` field in `package.json` linked to these files. This is an anti-pattern and was rightfully flagged by the ["Are the Types Wrong?"](https://arethetypeswrong.github.io/) tool as ["masquerading as CJS"](https://github.com/arethetypeswrong/arethetypeswrong.github.io/blob/main/docs/problems/FalseCJS.md). All of the ATTW checks now pass.

Reviewer 2: > All of the ATTW checks now pass

This reminds me, I see we aren't running `attw` as part of the `test` script for this package. That's something we're doing for the module template, but not for `utils`, so it's understandable. We can do this in a new PR after the release, so not a big deal, but thought I'd note it.

Reviewer: What do you think about just removing that last sentence? The rest seems like relevant, useful information.

```suggestion
  - Previously, this package shipped with only one variant of type declaration files, and these files were only CommonJS-compatible, and the `exports` field in `package.json` linked to these files. This is an anti-pattern and was rightfully flagged by the ["Are the Types Wrong?"](https://arethetypeswrong.github.io/) tool as ["masquerading as CJS"](https://github.com/arethetypeswrong/arethetypeswrong.github.io/blob/main/docs/problems/FalseCJS.md).
```

Reviewer 2: Okay, I'm good with that.
